pkg/asset/machines/baremetal/hosts.go

```diff
@@ -29,8 +32,27 @@ type HostSettings struct {
 	NetworkConfigSecrets []corev1.Secret
 }
 
-func createNetworkConfigSecret(host *baremetal.Host) corev1.Secret {
-	return corev1.Secret{
+func convertNetworkConfigToYaml(host *baremetal.Host) ([]byte, error) {
+	var yamlNetworkConfig []byte
+
+	jsonNetworkConfig, err := json.Marshal(host.NetworkConfig)
+	if err == nil {
+		yamlNetworkConfig, err = yaml.JSONToYAML([]byte(jsonNetworkConfig))
+	}
```

Comment:
Why do we need to marshal to JSON and then convert the JSON to YAML as opposed to directly marshaling to YAML?

Reply:
Thanks for the suggestion @staebler even though the implementation was the same I could save few lines of code